### ISSUE TITLE
Appfolders on the dock

### ISSUE THREAD

**Issue by CharoSW** [micheleg/dash-to-dock#332]
Is it possible to have AppFolders on the dock, or a similar custom implementation?

**Comment by micheleg** [comment]
Thanks for your suggestion. Would you mind providing more details on the motivation and possible implementations (mockups, example of implementations in other dock/OS) of this idea? Somehow the _show apps_ icon is already a consistent and logical way to provide access to additional applications.

**Comment by lucasvieites** [comment]
Hi, I would like to adda +1 to this. This feature is useful to group types of applications on folders, much like the "Drawers" did in Gnome 2. I'm attaching a raw mockup of the intended functionality, in this example the user has a folder with his favorite games for quick access instead of having to look for them in the Applications. I used it a to have "Design", "Programming", "Games", etc. folders.
I hope this is a feasible feature.
Thank you and kind regards,
Lucas
![dash-to-dock-drawers-mockup](https://user-images.githubusercontent.com/1646239/36350267-fb244818-1495-11e8-83a1-1f61032eae85.png)

**Comment by Huluti** [comment]
Could be a good idea. I have many apps in my dock and group some within folders could be a good idea (dev tools,...). I can view all the apps on a monitor but not on my laptop screen.

**Comment by lvanz** [comment]
really need this...

**Comment by enx1998** [comment]
Hi, this would be a very helpfull feature beacuse it would permits to organize icons in the dock allowing us to launch applications without the need to open full screen apps list. The same behaviour is already present in Plank (Elementary OS) and in Dock of MacOS.
Thanks for your great work.

**Comment by benjamin-wilson** [comment]
My #1 want in dash to dock.

**Comment by danielstuart14** [comment]
This would be awesome

**Comment by MagellanTX** [comment]
+1, This would be very handy to organize the apps!

**Comment by esa1975** [comment]
I tend to use quite a lot of favorites on the dock and would love to see this enhancement. Thank you for continuing to provide what I consider to be an essential part of the GNOME desktop experience. I greatly appreciate it!

**Comment by vmavromatis** [comment]
This should be THE top feature needed for the dash-to-dock IMHO. +1 from me

**Comment by unix2dossss** [comment]
we still need this!

**Comment by hazrpg** [comment]
Just to add to this, this is a common feature in Android too.
I've temporarily worked around this by updating existing .desktop files to add multiple into the right-click menu... but its far from ideal, and not a rock solid implementation because updates can easily mess with the files (even though they are in `~/.local/share/applications` which in theory should be a local reference of the original and an override).

My list of regular applications that I use for work makes the list become a scroll bar of icons - which is good, but it would be better to group by task. E.g.:

Work - PhpStorm, PyCharm, DBeaver, Meld, Postman, Google Meet (pwa), Google Calendar (pwa), etc
Games - Steam, Lutris, MultiMC 5
Browsers - Firefox, Opera, Chrome, Edge, Chromium, Brave
Utilities - Calculator, PrusaSlicer, Gnome Text Editor, GIMP, OBS, qBT

And so many more.

I've already got my icons set to as low as it can go (16px I think it is), any smaller and it would totally be unusable, but ideally I would like it more like 18px or 20px... but the scrolling would be painful.

Have you got a way to help fund a feature like this from being implemented? Because I'm sure folk would happily donate to get this in.

**Comment by orangesunny** [comment]
This feels like an essential feature for me

**Comment by yhojann-cl** [comment]
If it's essential and has been requested since 2016 and there's still no sign that it will be implemented or planned, most likely we'll never see it until some fork of the plugin is implemented in distributions, but not the original due to lack of feature maintenance.

**Comment by rasyidrafi** [comment]
what use case of this? i didnt seem find any use case of? did pinning all of those applications aren't enough?

**Comment by yar0d** [comment]
Gnome already manage folders to group applications, see below " Bureautique " and " utilitaires " :

<img width="1052" height="541" alt="Image" src="https://github.com/user-attachments/assets/da3f81ae-4a0f-45d0-a78e-e50256c9f8ad" />

A good use case is described there : https://github.com/micheleg/dash-to-dock/issues/332#issuecomment-3023183171

It will make it easy to group applications and so reduce de size of dock. An example is a group "office" containing all libreoffice applications. Another example is one group "design" that will contains inkscape, gimp, and other image edition tools. It's pretty similar to android folder.

**Comment by hazrpg** [comment]
For me, the use case is separating tasks out. The more icons you use regularly per day the bigger the scroll is on the bar. Which gets old really quick.

An alternative would be to have the dash update based on the workspace you're in so that you can separate work from personal. But that seems like it would be a nightmare to set up. 

Folders work better because anyone can set up a folder system that works for them.

**Comment by yhojann-cl** [comment]
> what use case of this? i didnt seem find any use case of? did pinning all of those applications aren't enough?

No, for productive work, it's not always enough when you need quick access to different things within the system, whether it's applications, folders, or even files that aren't applications or folders, such as .appimage files. All commercial operating systems, whether macOS or Windows, offer customization options for quick directory access. Perhaps not everyone understands whether this is truly necessary, but the market and widespread adoption have proven that it is. I've read countless articles about Ubuntu's upcoming releases, where they claim to focus on user experience, but beyond the design, we only see more restrictions instead of greater flexibility in customization.
